### PR TITLE
Resolve several minor codacy-reported issues

### PR DESCRIPTION
**Pull request by GernotMaier**
This is a minor pull request which solves several issues reported by codacy.

Main changes: 
- reduced complexity of `general._validateAndConvertValue()` and addition of unit tests (I actually managed to understand the method only with help of the tests).
- yaml loaders are now always safe_loaders

Issue #257

### CODE REVIEW

**Comment by GernotMaier** [comment]
Something for next week....

**Review by orelgueta — COMMENTED** [review]
Minor comments, one use of print which blocks approval

File: simtools/util/general.py [hunk 1]
```diff
@@ -143,92 +143,164 @@ def validateConfigData(configData, parameters):
...
-
-        # Checking units and converting them, if needed.
-        valueWithUnits = list()
-        for arg, unit in zip(value, parUnit):
-            # In case a entry is None, None should be returned.
-            if unit is None or arg is None:
-                valueWithUnits.append(arg)
-                continue
+    if "unit" not in parInfo.keys():
+        print("NOOOOOUNITS")
```
[the remark this answers is not in the document]
Comment (GernotMaier) [reply simtools/util/general.py:300]: I guess now you see how informative my debug statements are. This should go.

File: simtools/camera_efficiency.py [hunk 2]
```diff
@@ -4,9 +4,9 @@
 from collections import defaultdict
 from pathlib import Path
 
+import astropy.io.ascii
```
[the remark this answers is not in the document]
Comment (GernotMaier) [reply simtools/camera_efficiency.py:7]: I've changed it to `import astropy.io.ascii`. 

`asc` does not make it clear when reading the code that this is ascii.

**Comment by GernotMaier** [comment]
I removed the print statements, so I guess it is approved and I can merge.

**Pull request merged**